internal/proto/volumegroup.proto

```diff
@@ -0,0 +1,182 @@
+  // This field is OPTIONAL.
+  // To support the creation of an empty group, this list can be empty.
+  // However, this field is not empty in the following cases:
+  // - Response from ListVolumeGroups or ControllerGetVolumeGroup if the
+  //   VolumeGroup is not empty.
+  // - Response from ModifyVolumeGroupMembership if the
+  //   VolumeGroup is not empty after modification.
+  repeated string volume_ids = 3;
+}
+// DeleteVolumeGroupRequest holds the required information to
+// delete a volume group
+message DeleteVolumeGroupRequest {
+  // The ID of the volume group to be deleted.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // delete request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// DeleteVolumeGroupResponse is the response to send when
+// volumeGroup is successfully deleted.
+message DeleteVolumeGroupResponse {
+}
+// ModifyVolumeGroupMembershipRequest holds the required
+// information to modify a volume group
+message ModifyVolumeGroupMembershipRequest {
+  // The ID of the volume group to be modified.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Specify volume_ids that will be in the modified volume group.
+  // This list will be compared with the volume_ids in the existing
+  // group.
+  // New ones will be added and missing ones will be removed.
+  // If no volume_ids are provided, all existing volumes will
+  // be removed from the group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 2;
+
+  // Secrets required by the driver to complete the
+  // modification request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+
+  // params passed to the plugin to modify the volume group
+  // or to modify the volumes in the group
+  // This field is OPTIONAL.
+  map<string, string> parameters = 5;
+}
+// ModifyVolumeGroupMembershipResponse is the response to
+// send when volumeGroup is successfully modified.
+message ModifyVolumeGroupMembershipResponse {
+  // Contains all attributes of the modified volume group.
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+// ControllerGetVolumeGroupRequest holds the required
+// information to get information on volume group
+message ControllerGetVolumeGroupRequest {
+  // The ID of the volume group to fetch current volume group
+  // information for.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // get request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// ControllerGetVolumeGroupResponse is the response to
+// send when volumeGroup information was successfully gathered.
+message ControllerGetVolumeGroupResponse {
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+// ListVolumeGroupsRequest holds the required
+// information to get information on list of volume groups.
+message ListVolumeGroupsRequest {
```

Comment:
Are you planning to use this function? No need to include functions that you won't use anyway.

Reply:
I am not but as per the above comments, I included it.
I will remove it in my controller PR, or shall I remove it here itself?

Reply:
I do not see a point in including anything that won't be used, that can only cause confusion. Do not add things that will be removed in a follow-up commit/PR.

The files under `internal/proto/` are only for the communication between the controller and sidecar, it should be as simple and minimal as possible.

Reply:
Updated the code to remove the ListVolumeGroups RPC.